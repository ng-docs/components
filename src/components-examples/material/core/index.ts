import {NgModule} from '@angular/core';
import {FormsModule} from '@angular/forms';
import {MatButtonModule} from '@angular/material/button';
import {MatCheckboxModule} from '@angular/material/checkbox';
import {MatRippleModule} from '@angular/material/core';
import {MatInputModule} from '@angular/material/input';
import {ElevationOverviewExample} from './elevation-overview/elevation-overview-example';
import {RippleOverviewExample} from './ripple-overview/ripple-overview-example';

export {ElevationOverviewExample, RippleOverviewExample};

const EXAMPLES = [ElevationOverviewExample, RippleOverviewExample];

@NgModule({
  imports: [MatButtonModule, MatCheckboxModule, MatInputModule, MatRippleModule, FormsModule],
  declarations: EXAMPLES,
  exports: EXAMPLES,
})
export class CoreExamplesModule {}
