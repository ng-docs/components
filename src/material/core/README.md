Core library code for other `@angular/material` components.

`@angular/material` 组件的核心库代码。