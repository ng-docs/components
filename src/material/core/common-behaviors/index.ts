/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

export {
  MatCommonModule,
  MATERIAL_SANITY_CHECKS,
  SanityChecks,
  GranularSanityChecks,
} from './common-module';

// Note: These need to be exposed privately for cross-package type inference. e.g. if the
// experimental package uses a mixin, TS will try to write an explicit type reference that
// is equivalent to e.g. `CanColorCtor`. For this it needs these two helpers as otherwise it
// would generate a deep cross-package import that breaks in the NPM package output.
export {
  Constructor as _Constructor,
  AbstractConstructor as _AbstractConstructor,
} from './constructor';

export {CanDisable, mixinDisabled} from './disabled';
export {CanColor, mixinColor, ThemePalette} from './color';
export {CanDisableRipple, mixinDisableRipple} from './disable-ripple';
export {HasTabIndex, mixinTabIndex} from './tabindex';
export {CanUpdateErrorState, mixinErrorState} from './error-state';
export {HasInitialized, mixinInitialized} from './initialized';
