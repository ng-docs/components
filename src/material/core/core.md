The `@angular/material/core` package contains common components and services used by multiple other
components in the library. See the API tab for a listing of components and classes available.
