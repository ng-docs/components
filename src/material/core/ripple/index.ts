/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import {NgModule} from '@angular/core';
import {MatCommonModule} from '../common-behaviors/common-module';
import {MatRipple} from './ripple';

export * from './ripple';
export * from './ripple-ref';
export * from './ripple-renderer';

@NgModule({
  imports: [MatCommonModule],
  exports: [MatRipple, MatCommonModule],
  declarations: [MatRipple],
})
export class MatRippleModule {}
