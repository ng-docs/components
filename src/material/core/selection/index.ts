/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import {NgModule} from '@angular/core';
import {MatPseudoCheckbox} from './pseudo-checkbox/pseudo-checkbox';
import {MatCommonModule} from '../common-behaviors/common-module';

@NgModule({
  imports: [MatCommonModule],
  exports: [MatPseudoCheckbox],
  declarations: [MatPseudoCheckbox],
})
export class MatPseudoCheckboxModule {}

export * from './pseudo-checkbox/pseudo-checkbox';
