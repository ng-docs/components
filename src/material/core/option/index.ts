/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {MatRippleModule} from '../ripple/index';
import {MatPseudoCheckboxModule} from '../selection/index';
import {MatCommonModule} from '../common-behaviors/common-module';
import {MatOption} from './option';
import {MatOptgroup} from './optgroup';

@NgModule({
  imports: [MatRippleModule, CommonModule, MatCommonModule, MatPseudoCheckboxModule],
  exports: [MatOption, MatOptgroup],
  declarations: [MatOption, MatOptgroup],
})
export class MatOptionModule {}

export * from './option';
export * from './optgroup';
export * from './option-parent';
