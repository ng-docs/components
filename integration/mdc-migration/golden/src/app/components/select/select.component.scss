.mat-mdc-select {
  padding: 4px;
}

/* TODO(mdc-migration): The following rule targets internal classes of select that may no longer apply for the MDC version. */
.select-example ::ng-deep.mat-select-value {
  color: darkblue;
}


.mat-mdc-select-panel {
  padding: 4px;

  .mat-mdc-option {
    padding: 4px;
  }
}
