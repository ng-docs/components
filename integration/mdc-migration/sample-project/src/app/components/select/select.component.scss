.mat-select {
  padding: 4px;
}

.select-example ::ng-deep.mat-select-value {
  color: darkblue;
}


.mat-select-panel {
  padding: 4px;

  .mat-option {
    padding: 4px;
  }
}
