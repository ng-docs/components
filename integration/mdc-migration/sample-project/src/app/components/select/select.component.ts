import {Component} from '@angular/core';

interface Food {
  value: string;
  viewValue: string;
}

@Component({
  selector: 'select-example',
  templateUrl: './select.component.html',
  styleUrls: ['./select.component.scss'],
})
export class SelectComponent {
  foods: Food[] = [
    {value: 'steak-0', viewValue: 'Steak'},
    {value: 'pizza-1', viewValue: 'Pizza'},
    {value: 'tacos-2', viewValue: 'Tacos'},
  ];
}
