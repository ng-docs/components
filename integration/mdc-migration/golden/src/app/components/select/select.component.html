<h2>Select example</h2>
<mat-form-field class="select-example" appearance="fill">
  <mat-label>Favorite food</mat-label>
  <mat-select>
    <mat-option *ngFor="let food of foods" [value]="food.value">
      {{food.viewValue}}
    </mat-option>
  </mat-select>
</mat-form-field>
